### PR TITLE
fix: url.parse expects req.url not req

### PR DESCRIPTION
**Pull request by Qard**
The `parseurl` module receives a request object while `url.parse` expects to receive the url string. This should fix that. I'm not sure how to test this though, as it loads the url module as a fallback when first running the module. An alternative would maybe be to just add `parseurl` to the dependencies and _always_ use it. 🤔 

### Checklist

- [x] Implement code
- [ ] Add tests

### CODE REVIEW

File: lib/instrumentation/express-utils.js [hunk 1]
```diff
@@ -4,7 +4,8 @@ var parseUrl
 try {
   parseUrl = require('parseurl')
 } catch (e) {
-  parseUrl = require('url').parse
+  const url = require('url')
+  parseUrl = req => url.parse(req.url)]
```
Comment (watson) [inline lib/instrumentation/express-utils.js]:
```suggestion
  parseUrl = req => url.parse(req.url)
```

**Comment by watson** [comment]
Good catch! [Our tests](https://github.com/elastic/apm-agent-nodejs/blob/0aaa7e0ad792d5a4ce94e0b0f7421c66a7407641/test/config.js#L560-L590) should actually already have caught this as far as I can see. We should look into why they didn't as part of this PR 🤔 

I don't like adding `parseurl` as a dependency as it's messing with the `req` object in a way normal users might not like.

**Comment by watson** [comment]
Found the reason why our tests didn't catch this:

image

Because we already have `connect`, `express`, `finalhandler`, and `koa` as devDependencies, we indirectly have `parseurl` installed when running our tests and therefore our tests will not throw an error and will instead use the fallback when requiring the `parseurl` module. So that code-path isn't tested at all.

Not sure how we can test this without literally running `rm -fr node_modules/parseurl` ... maybe that's what you meant and I just didn't get it 😅

**Comment by watson** [comment]
The failing tests are just because Travis is being difficult this morning. Let's just ignore those

**Comment by watson** [comment]
After giving it some thought, I think it's best to just merge this as is so we can get it released today. I'll create a separate issue to update the tests

**Comment by Qard** [comment]
Yes, the indirect dev dependency is why I wasn't sure if there was a good way to test this and why I felt the existing tests didn't cover it.

**Pull request merged**